### PR TITLE
Physical device selection process is redesigned and enhanced

### PR DESCRIPTION
**Pull request by MoonDweller29**
Now physical device selection is performed via HardwareManager

### CODE REVIEW

File: include/magma/app/Option.hpp [hunk 1]
```diff
@@ -0,0 +1,56 @@
+/**
+ * @file Option.hpp
```
Comment (Nastormo) [inline include/magma/app/Option.hpp:2]: Why use one hpp file? Do you think we will don't expand functionality?

File: include/magma/vk/logicalDevice.h [hunk 2]
```diff
@@ -25,9 +24,10 @@ class LogicalDevice
 public:
 
 
-    LogicalDevice(const PhysicalDevice &physicalDevice);
+    LogicalDevice(const PhysicalDevice &physicalDevice, const std::vector<const char*> &deviceExtensions);
     VkDevice handler() const { return device; }
-    const VkPhysicalDevice &physDevice() const { return physicalDevice; }
+    VkPhysicalDevice getVkPhysDevice() const { return _physDevice.c_device(); }
+    const PhysicalDevice &physDevice() const { return _physDevice; }
```
Comment (Nastormo) [inline include/magma/vk/logicalDevice.h:30]: How we markmark get function?

**Review by Nastormo — CHANGES_REQUESTED** [review]

File: include/magma/vk/swapChain.h [hunk 3]
```diff
@@ -1,5 +1,5 @@
 #pragma once
-#include "physicalDevice.h"
+#include "magma/vk/physicalDevice/PhysicalDevice.h"
```
Comment (Nastormo) [inline include/magma/vk/swapChain.h:2]: Move down this line please

File: src/magma/vk/physicalDevice/PhysicalDevice.cpp [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+        ss << " - Surface Support";
+    }
+    ss << " - " << queueFamily.queueCount;
+
+    return ss.str();
+}
+
+QueueFamilyIndices PhysicalDevice::findQueueFamilies(
+        vk::PhysicalDevice device, vk::SurfaceKHR surface
+) {
```
Comment (Nastormo) [inline src/magma/vk/physicalDevice/PhysicalDevice.cpp:36]: May move name function and parameters on one line?

File: include/magma/vk/physicalDevice/HardwareManager.h [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+public:
+    HardwareManager(vk::Instance instance);
+
+    std::vector<PhysicalDevice> acquirePhysicalDevicesList();
+    std::multimap<int, PhysicalDevice> findSuitableDevices(const DeviceRequirements &requirements);
+    PhysicalDevice selectBestSuitableDevice(const DeviceRequirements &requirements);
+private:
+    vk::Instance _instance;
+
+    bool isDeviceSuitable(const PhysicalDevice &device, const DeviceRequirements &requirements);
```
Comment (Nastormo) [inline include/magma/vk/physicalDevice/HardwareManager.h:20]: Test commit

**Review by Nastormo — CHANGES_REQUESTED** [review]

File: include/magma/vk/swapChain.h (hunk 3 above)
Comment (MoonDweller29) [reply include/magma/vk/swapChain.h:2]: This part will be refactored with the SwapChain moving to hpp

File: include/magma/app/Option.hpp (hunk 1 above)
Comment (MoonDweller29) [reply include/magma/app/Option.hpp:2]: It's a file for a template class

File: include/magma/vk/logicalDevice.h (hunk 2 above)
Comment (MoonDweller29) [reply include/magma/vk/logicalDevice.h]: changed to getPhysDevice()

File: src/magma/vk/physicalDevice/PhysicalDevice.cpp (hunk 4 above)
Comment (MoonDweller29) [reply src/magma/vk/physicalDevice/PhysicalDevice.cpp]: done

**Review by Nastormo — APPROVED** [review]

**Pull request merged**